module/VuFind/src/VuFind/Sitemap/Generator.php

```diff
@@ -347,13 +370,18 @@ protected function getIdsFromBackendUsingCursorMark(Backend $backend, $cursorMar
                 'cursorMark' => $cursorMark
             ]
         );
-        $raw = $connector->search($params);
-        $result = json_decode($raw);
-        $ids = [];
-        $nextOffset = $result->nextCursorMark;
-        foreach ($result->response->docs ?? [] as $doc) {
-            $ids[] = $doc->$key;
+        $results = $this->searchService->search(
+            $backend->getIdentifier(),
+            new Query('*:*'),
+            0,
+            $this->countPerPage,
+            $params,
+            'getids'
```

Comment:
If our goal is to standardize this so that standard listeners fire on sitemap generation, using this non-standard context seems likely to cause issues, since most listeners are going to be expecting `search` in this context. I understand that you need a way to flag this behavior for the DeduplicationListener; I just wonder if we need a better mechanism to accomplish that. For example, would stuffing something into the ParamBag potentially be less disruptive? Or do we need some new way of tagging requests independently of the ParamBag?

Reply:
As I see it this is a new context (fetching ID's instead of doing a normal search), and using 'getids' is more back-compatible since the listeners weren't firing before either. Any existing listener wouldn't know to check a new flag in parameters. Yes, the downside is that existing listeners need to be modified for getids, but that makes sure they're also modified only if appropriate.

Reply:
That makes sense, but if we go that route, I wonder if we should go a little further and actually implement a getids method on the backend. Running a search under an alias like this seems like a bit of a hack, and I think it makes for somewhat non-obvious code. We could implement this as an optional Feature interface, like RetrieveBatchInterface, etc., so we don't have to implement it across all the backends.

It's entirely possible this is more work than it's worth -- I imagine this could end up duplicating a lot of code and/or creating IdCollections and IdCollectionFactories parallel to RecordCollections, etc. -- but I just want to talk it through and see if we can do this in a more robust and straightforward way before settling on this approach.

Reply:
I added the feature interface and a getIds method that uses a normal search as fallback. I didn't tie the different collection type to it, since it can just as well be used with the normal record collection if need be. What do you think?

Reply:
Sounds reasonable to me; I'll give this a closer review next week when I have a bit more time!